src/xtgeo/surface/_regsurf_gridding.py

```diff
@@ -39,46 +162,1551 @@ def points_gridding(self, points, method="linear", coarsen=1):
+                            dt = (-dy_ds * rx + dx_ds * ry) / det
+
+                            s += ds
+                            t += dt
+
+                            # Clamp to valid range
+                            s = np.clip(s, 0, 1)
+                            t = np.clip(t, 0, 1)
+
+                        # Calculate bilinear weights
+                        w00 = (1 - s) * (1 - t)
+                        w10 = s * (1 - t)
+                        w01 = (1 - s) * t
+                        w11 = s * t
+
+                        # Current interpolated value
+                        z_current = (
+                            w00 * znew[i0, j0]
+                            + w10 * znew[i1, j0]
+                            + w01 * znew[i0, j1]
+                            + w11 * znew[i1, j1]
+                        )
+
+                        # Calculate and distribute adjustment
+                        z_adjust = z - z_current
+                        znew[i0, j0] += w00 * z_adjust
+                        znew[i1, j0] += w10 * z_adjust
+                        znew[i0, j1] += w01 * z_adjust
+                        znew[i1, j1] += w11 * z_adjust
+
+                        n_enforced += 1
+                        break
+
+                if found_cell:
+                    break
+
+        if not found_cell:
+            n_outside += 1
+
+    logger.debug(
+        "Enforced exact values for %d/%d points using bilinear adjustment "
+        "(%d points outside grid)",
+        n_enforced,
+        n_tested,
+        n_outside,
+    )
+    return n_enforced
+
+
+def points_kriging_gridding(
+    self: RegularSurface,
+    points: Points,
+    variogram_model: str = "Gaussian",
+    variogram_parameters: dict[str, Any] | None = None,
+    len_scale: float | tuple[float, float] | None = None,
+    range_value: float | tuple[float, float] | None = None,
+    nugget: float = 0.0,
+    krige_type: Literal["ordinary", "simple"] = "ordinary",
+    mean: float | None = None,
+    coarsen: int = 1,
+    max_points: int = 500,
+    radius: float | None = None,
+    exact: bool = False,
+    angle: float | None = None,
+) -> None:
+    """
+    Do kriging gridding from a points data set using GSTools.
+
+    Kriging provides optimal spatial interpolation based on spatial correlation
+    structure (variogram). Good for sparse, spatially correlated data.
+
+    Args:
+        points: Points object containing x, y, z coordinates
+        variogram_model: Variogram/Covariance model. Options:
+            - 'Gaussian': Gaussian model (smooth, good for continuous phenomena)
+            - 'Exponential': Exponential model (good for geology)
+            - 'Spherical': Spherical model (classic geostatistics)
+            - 'Matern': Matérn model (flexible smoothness)
+            - 'Stable': Stable model (generalization of Gaussian, also called
+              'General exponential')
+            - 'Linear': Linear model
```

Comment:
I would keep these options lowercase which is a more common practice and in line with most other xtgeo functions where strings are input.
`capitalize` is applied to the strings before extracting the variogram model class from `gstools` in `_create_variogram_model` anyway 🙂